examples/server/server.cpp

```diff
@@ -512,7 +524,7 @@ struct llama_server_context
     bool all_slots_are_idle = false;
     bool add_bos_token      = true;
 
-    int32_t id_gen;
+    std::atomic<int32_t> id_gen;
```

Comment:
Something bothered me about id_gen not being atomic explicitly. This should make it more robust.

Reply:
> This should make it more robust.

We should not make things atomic without a very good reason.
Is there any reason in this case?

Reply:
id_gen should not be atomic. Its use is always guarded by mutex_tasks.

Reply:
It was not obvious to me that id_gen was guarded by mutex_tasks. It's correct that it was, though. I've reverted to the old behavior.